### ISSUE TITLE
Colurful Console - Support colors in the embedded Console

### ISSUE THREAD

**Issue by ITAYC0HEN** [radareorg/cutter#705]
Make Console widget prettier by adding a support for colors.

Currently it is only B&W:

![image](https://user-images.githubusercontent.com/20182642/45393150-e31cd780-b62a-11e8-8fce-f4a22dae58f0.png)

**Comment by tkhurana96** [comment]
I would like to give it a shot...

**Comment by tkhurana96** [comment]
@ITAYCOHEN, I will get started with building the project using the link you shared

**Comment by ITAYC0HEN** [comment]
Hey @tkhurana96 :)
How things are going? Just following up to see that everything is ok.
If you need some help, don't hesitate - we are here :D

**Comment by tkhurana96** [comment]
Will start really soon working on it😀

**Comment by tkhurana96** [comment]
Hi all, I was able to build the project succesfully, and I have gone through the code too. My finding till now is that radare2 is not sending the ANSI color sequences for commands. I have reached this portion of code: 

```
R_API char *r_core_cmd_str(RCore *core, const char *cmd) {
	const char *static_str;
	char *retstr = NULL;
	r_cons_push ();
	if (r_core_cmd (core, cmd, 0) == -1) {
		//eprintf ("Invalid command: %s\n", cmd);
		return NULL;
	}
	r_cons_filter ();
	static_str = r_cons_get_buffer ();
	retstr = strdup (static_str? static_str: "");
	r_cons_pop ();
	return retstr;
}
```

@ITAYC0HEN Can you guide me what does `r_cons_push` does ? I tried looking into it, but couldn't get a sense of it. I am basically trying to find out how are commands(that are sent from cutter) being parsed and executed by radare. Because the return value of the above the function is the output of a command being executed, we can find why radare doesn't puts ANSI codes in the output

**Comment by xarkes** [comment]
For radare2 to send the ANSI color, you have to make sure that the variable `scr.color` is different than `0`.
Some commands overwrite it by default (setting it to 0), using the TempConfig class.

**Comment by tkhurana96** [comment]
@xarkes Thanks, I tried setting `scr.color` to 2 and it works :smile: !! So, I was able to enable colors in radare2 output, here is the PR #823 .

**Comment by xarkes** [comment]
Yeah I think that's because the console doesn't support ANSI codes.
You might want to try with `scr.html = true`, I think this widget can handle html.

**Comment by Adikso** [comment]
I think that it should be closed #823
